src/styles/block.scss

```diff
@@ -198,6 +198,122 @@
 	}
 }
 
+// Buttons.
+.stk-block {
+	.stk-button {
+		justify-content: center;
+		position: relative;
+		display: flex;
+		align-items: center;
+		column-gap: 4px;
+		text-decoration: none;
+		// make sure that the anchor tag's width matches the parent's width.
+		width: 100%;
+
+		&::before,
+		&::after {
+			content: "";
+			position: absolute;
+			top: 0;
+			left: 0;
+			pointer-events: none;
+			border-radius: inherit;
+		}
+
+		&::before {
+			bottom: 0;
+			right: 0;
+			// Borders should be on top of the ::after pseudoelement.
+			z-index: 2;
+		}
+
+		&::after {
+			width: 100%;
+			height: 100%;
+			opacity: 0;
+			z-index: 1;
+		}
+
+		.stk--inner-svg {
+			display: flex;
+			svg:last-child {
+				height: 1em;
+				width: 1em;
+				&,
+				:is(g, path, rect, polygon, ellipse) {
+					fill: cssvar(button-text-color);
+				}
+			}
+		}
+
+		> * {
+			z-index: 2;
+		}
+	}
+}
+
+.stk-block {
+	&.is-style-pill {
+		a.stk-button {
+			width: 64px;
+		}
+	}
+	&.is-style-plain {
+		a.stk-button {
+			.stk--inner-svg {
+				color: cssvar(button-background-color);
+			}
+			&:hover {
+				.stk-button__inner-text {
+					color: cssvar(button-background-color);
+				}
+			}
+
+			> .stk-button__inner-text {
+				color: cssvar(button-background-color);
+			}
+		}
+	}
+	&.is-style-ghost {
+		a.stk-button {
+			&::before {
+				border-width: cssvar(button-ghost-border-width);
+				border-color: cssvar(button-background-color);
+			}
+			.stk--inner-svg {
+				color: cssvar(button-background-color);
+			}
+
+			&:hover {
+				.stk-button__inner-text {
+					color: cssvar(button-background-color);
+				}
+			}
+
+			> .stk-button__inner-text {
+				color: cssvar(button-background-color);
+			}
+		}
+	}
+	&.is-style-ghost,
+	&.is-style-plain {
+		a.stk-button > .stk--svg-wrapper {
+			.stk--inner-svg {
+				color: cssvar(button-background-color);
+				&,
+				:is(g, path, rect, polygon, ellipse) {
+					fill: cssvar(button-background-color);
+				}
+			}
+		}
+	}
+}
```

Comment:
Please move all `cssvar` and design system related styles in `block-design-system.scss`